JavaScript/FightingGame/app.js

```diff
@@ -0,0 +1,400 @@
+class CharacterImage {
+  constructor(characterName) {
+    this.element = document.createElement("img");
+    this.animations = {
+      idle: null,
+      attack1: null,
+      attack2: null,
+      attack3: null,
+      defend: null,
+      defeat: null,
+      damage: null,
+    };
+
+    return this.calcAnimations({ characterName });
+  }
+
+  async calcAnimations({ baseUrl = "./assets/characters", characterName }) {
+    const animationNames = Object.keys(this.animations);
+    const durations = animationNames.map((animationName) => {
+      return this.#calcAnimationDuration({
+        baseUrl,
+        characterName,
+        animationName,
+      });
+    });
+
+    await Promise.all(durations);
+    return this;
+  }
+
+  loadImage(animationName) {
+    const blob = new Blob([this.animations[animationName].arrayBuffer], {
+      type: "image/gif",
+    });
+    const imageUrl = URL.createObjectURL(blob);
+    this.element.src = imageUrl;
+    this.element.alt = animationName;
+  }
+
+  async #calcAnimationDuration({ baseUrl, characterName, animationName }) {
+    return fetch(`${baseUrl}/${characterName}/${animationName}.gif`)
+      .then((res) => res.arrayBuffer())
+      .then((ab) => {
+        return (this.animations[animationName] = {
+          duration: this.#isGifAnimated(new Uint8Array(ab)),
+          arrayBuffer: new Uint8Array(ab),
+        });
+      });
+  }
+
+  #isGifAnimated(uint8) {
+    let duration = 0;
+    for (let i = 0, len = uint8.length; i < len; i++) {
+      if (
+        uint8[i] == 0x21 &&
```

Comment:
we can use const for magic numbers

`const GIF_BLOCK_TYPE = 0x21;
const EXTENSION_INTRODUCER = 0xf9;
const GRAPHIC_CONTROL_LABEL = 0x04;
const TERMINATOR = 0x00;
`

`if (
            uint8[i] === GIF_BLOCK_TYPE &&
            uint8[i + 1] === EXTENSION_INTRODUCER &&
            uint8[i + 2] === GRAPHIC_CONTROL_LABEL &&
            uint8[i + 7] === TERMINATOR
        ) {`